### PR TITLE
streamingccl: support sst event in random stream client

### PR DESCRIPTION
**Pull request by gh-casper**
Previously random stream client lacks test coverage
for AddSSTable operation, this PR enables random stream
client to generate and validate random SSTableEvents.

This PR also cleans up the InterceptableStreamClient
to avoid unnecessary abstraction, making code cleaner.

Release justification: low risk, high benefit changes to 
existing functionality

Release note : None

### CODE REVIEW

File: pkg/ccl/streamingccl/streamclient/random_stream_client.go [hunk 1]
```diff
@@ -196,19 +193,89 @@ func (c randomStreamConfig) URL(table int) string {
...
+		sstMaker:                  fn,
+		tableDesc:                     tableDesc,
+		systemKVs:                     systemKVs,
+	}, nil
+}
+
+func (r *randomEventGenerator) generateNewEvent() streamingccl.Event {
+		var event streamingccl.Event
+		if r.numKVEventsSinceLastResolved == r.config.kvsPerCheckpoint &&
+			r.numKVEventsSinceLastResolved == r.config.sstsPerCheckpoint {
```
Comment (stevendanna) [inline pkg/ccl/streamingccl/streamclient/random_stream_client.go:238]: Should this be `numSSTEventsSinceLastResolved`?

**Review by gh-casper — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/85392)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @miretskiy, @samiskin, and @stevendanna)

___
*[`pkg/ccl/streamingccl/streamclient/random_stream_client.go` line 238 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/85392#-N8MUWUeFtDROWtnhJZk-r1-238:-N9EwloeAjHxizpRIOzA:bqt6dmh) ([raw file](https://github.com/cockroachdb/cockroach/blob/83e1ff8c57b5d8b6631e7e3d48c53cf3bfab06ec/pkg/ccl/streamingccl/streamclient/random_stream_client.go#L238)):*
<details><summary><i>Previously, stevendanna (Steven Danna) wrote…</i></summary><blockquote>

Should this be `numSSTEventsSinceLastResolved`?
</blockquote></details>

Done. I also changed the logic to generate SST events -- using probability instead of checking num of events generated which may cause bug and is not very intuitive. 

Decided to move the delrange x random_stream_client in another PR.

File: pkg/ccl/streamingccl/streamclient/random_stream_client.go [hunk 2]
```diff
@@ -594,31 +608,78 @@ func makeRandomKey(
...
+		sst := event.GetSSTable()
+		dataCopy := make([]byte, len(sst.Data))
+		copy(dataCopy, sst.Data)
+		dup = streamingccl.MakeSSTableEvent(roachpb.RangeFeedSSTable{
+			Data:    dataCopy,
+			Span:    sst.Span.Clone(),
+			WriteTS: sst.WriteTS,
+		})
+	default:
+		panic("unsopported event type")
```
Comment (stevendanna) [inline pkg/ccl/streamingccl/streamclient/random_stream_client.go:640]:
```suggestion
		panic("unsupported event type")
```

**Review by stevendanna — APPROVED** [review]
Thanks for adding this! I gave it a once over and it looks good to me.

**Pull request merged**